docs/src/FAQ.md

```diff
@@ -76,3 +76,105 @@ ChainRulesTestUtils.jl has some dependencies, so it is a separate package from C
 This means your package can depend on the light-weight ChainRulesCore.jl, and make ChainRulesTestUtils.jl a test-only dependency.
 
 Remember to read the section on [On writing good `rrule` / `frule` methods](@ref).
+
+## How do chain rules work for complex functions?
+
+`ChainRules.jl` follows the convention that `frule` applied to a function ``f(x + i y) = u(x,y) + i v(x,y)`` with perturbation ``\Delta x + i \Delta y`` returns the value and
+```math
+\tfrac{\partial u}{\partial x} \, \Delta x + \tfrac{\partial u}{\partial y} \, \Delta y + i \, \Bigl( \tfrac{\partial v}{\partial x} \, \Delta x + \tfrac{\partial v}{\partial y} \, \Delta y \Bigr)
+.
+```
+Similarly, `rrule` applied to the same function returns the value and a pullback which, when applied to the adjoint ``\Delta u + i \Delta v``, returns
+```math
+\Delta u \, \tfrac{\partial u}{\partial x} + \Delta v \, \tfrac{\partial v}{\partial x} + i \, \Bigl(\Delta u \, \tfrac{\partial u }{\partial y} + \Delta v \, \tfrac{\partial v}{\partial y} \Bigr)
+.
+```
+If we interpret complex numbers as vectors in ``\mathbb{R}^2``, then these rules correspond to left- and right-multiplication with the Jacobian of ``f(z)``, i.e. `frule` corresponds to
+```math
+\begin{pmatrix}
+\tfrac{\partial u}{\partial x} \, \Delta x + \tfrac{\partial u}{\partial y} \, \Delta y
+\\
+\tfrac{\partial v}{\partial x} \, \Delta x + \tfrac{\partial v}{\partial y} \, \Delta y
+\end{pmatrix}
+=
+\begin{pmatrix}
+\tfrac{\partial u}{\partial x} & \tfrac{\partial u}{\partial y} \\
+\tfrac{\partial v}{\partial x} & \tfrac{\partial v}{\partial y} \\
+\end{pmatrix}
+\begin{pmatrix}
+\Delta x \\ \Delta y
+\end{pmatrix}
+
+```
+and `rrule` corresponds to
+```math
+\begin{pmatrix}
+\Delta u \, \tfrac{\partial u}{\partial x} + \Delta v \, \tfrac{\partial v}{\partial x}
+&
+\Delta u \, \tfrac{\partial u}{\partial y} + \Delta v \, \tfrac{\partial v}{\partial y}
+\end{pmatrix}
+=
+\begin{pmatrix}
+\Delta u & \Delta v
+\end{pmatrix}
+\begin{pmatrix}
+\tfrac{\partial u}{\partial x} & \tfrac{\partial u}{\partial y} \\
+\tfrac{\partial v}{\partial x} & \tfrac{\partial v}{\partial y} \\
+\end{pmatrix}
+.
+```
+The Jacobian of ``f:\mathbb{C} \to \mathbb{C}`` interpreted as a function ``\mathbb{R}^2 \to \mathbb{R}^2`` can hence be evaluated using either of the following functions.
+```
+function jacobian_via_frule(f,z)
+    fz,df_dx = frule((Zero(), 1),f,z)
+    fz,df_dy = frule((Zero(),im),f,z)
+    return [
+        real(df_dx)  real(df_dy)
+        imag(df_dx)  imag(df_dy)
+    ]
+end
+```
+```
+function jacobian_via_rrule(f,z)
+    fz, pullback = rrule(f,z)
+    _,du_dz = pullback( 1)
+    _,dv_dz = pullback(im)
+    return [
+        real(du_dz)  imag(du_dz)
+        real(dv_dz)  imag(dv_dz)
+    ]
+end
+```
+
+If ``f(z)`` is holomorphic, then the derivative part of `frule` can be implemented as ``f'(z) \, \Delta z`` and the derivative part of `rrule` can be implemented as ``\Delta f \,  \overline{f'(z)}``.
```

Comment:
This is incredibly pedantic, but the following would make this line up slightly better with the multidimensional case below:
```suggestion
If ``f(z)`` is holomorphic, then the derivative part of `frule` can be implemented as ``f'(z) \, \Delta z`` and the derivative part of `rrule` can be implemented as ``\overline{f'(z)} \, \Delta f``.
```

Reply:
Also, can you indicate here that `\overline{.}` means complex conjugate? Elsewhere we use `'` to mean conjugate and `\overline` to mean "adjoint".

Alternatively, since `\overline` is only used here several times for conjugation, perhaps it would be better to use `\operatorname{conj}`?

Reply:
Changed to `\operatorname{conj}`. This is probably the least confusing regardless of your background.